### PR TITLE
Fix incrementals support

### PR DESCRIPTION
**Pull request by Vlatombe**
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by Vlatombe** [comment]
```
[2021-11-18T12:14:28.611Z] Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?
[2021-11-18T12:14:28.611Z] Error: ZIP error: Error: Wrong commit hash in /project/scm/tag, expected ${scmTag}, got 2b4b4cb14a27d2182f260b395fc7703dda4c88a3 from https://ci.jenkins.io/job/Core/job/winstone/job/PR-175/1/artifact/**/*2b4b4cb14a27*/*2b4b4cb14a27*/*zip*/archive.zip
```

I fail to see what is still wrong... Any clue @jglick ?

**Comment by Vlatombe** [comment]
Ok, there is a conflict between flatten-maven-plugin and maven-shade-plugin

**Comment by jglick** [comment]
> Ok, there is a conflict between flatten-maven-plugin and maven-shade-plugin

Yes this is a well-known problem. Let me look up the fix…

**Comment by jglick** [comment]
5fdc61bb1cc2f3e1cc1ebf4dd25330cc97e30c0d (#81) should have taken care of this?

**Comment by Vlatombe** [comment]
Doesn't seem like it. In any case the resulting pom is wrong because it doesn't remove the shaded dependencies.

**Comment by jglick** [comment]
c007a70056783cdf63c39048b1e885bcf52fb528 by @olamy was not correct. Should have reverted to `${scmTag}` automatically—this is `mvn incrementals:reincrementalify` that should be run automatically by MRP.

**Comment by jglick** [comment]
FWIW enabling CD would avoid these sorts of problems, but my attempt was rejected in https://github.com/jenkins-infra/repository-permissions-updater/pull/1977.

**Comment by Vlatombe** [comment]
And yet, GH actions are used for CI of this repo...

**Comment by Vlatombe** [comment]
(anyway, I deployed a manual snapshot...)

**Comment by jglick** [comment]
https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/winstone/5.22-rc630.1d743cbb0b63/winstone-5.22-rc630.1d743cbb0b63.pom looks right to me @Vlatombe?

File: pom.xml [hunk 1]
```diff
@@ -186,10 +187,10 @@
   </pluginRepositories>
 
   <scm>
-    <connection>scm:git:git://github.com/jenkinsci/winstone.git</connection>
-    <developerConnection>scm:git:git@github.com:jenkinsci/winstone.git</developerConnection>
-    <url>http://github.com/jenkinsci/winstone</url>
-    <tag>HEAD</tag>
```
Comment (jglick) [inline pom.xml:192]: TBD how this got here to begin with

**Review by jglick — APPROVED** [review]

**Pull request merged**